src/lib/AllocationCbor.sol

```diff
@@ -0,0 +1,184 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.25;
+
+import "@ensdomains/buffer/contracts/Buffer.sol"; // for appendUint8
+import {CBOR} from "solidity-cborutils/contracts/CBOR.sol";
+import {CBORDecoder} from "filecoin-solidity/utils/CborDecode.sol";
+
+struct AllocationRequestData {
+    uint64 provider;
+    bytes dataCID;
+    uint64 size;
+    int64 termMin;
+    int64 termMax;
+    int64 expiration;
+}
+
+library AllocationCbor {
+    using Buffer for Buffer.buffer; // for appendUint8
+
+    function encodeData(
+        AllocationRequestData memory request
+    ) internal pure returns (bytes memory) {
+        CBOR.CBORBuffer memory buf = CBOR.create(64);
```

Comment:
I think we should be able to calculate the exact capacity we need based on length of allocationrequestdata, instead of hardcoding 64 (which almost definitely will be too small and will require very expensive resizing).

Reply:
Oops, scratch that, I forgot to delete this comment before submitting review.
64 is exactly the size that will be needed.